### DISCUSSION TITLE
Add SMPlayer extension to web browser

### DISCUSSION THREAD

**Discussion by trimechee** [smplayer-dev/smplayer#198] [Ideas]
Hello, il will be amazng if there in SMPlayer extension to send and play videos of web browsers in SMPlayer juste like tihs extension  which sends video and audio links to the media player. This way we can directly watch online streams in the SMPlayer with just one click ! Thank you.

https://microsoftedge.microsoft.com/addons/detail/open-in-vlc%E2%84%A2-media-player/hjfcjapkfahlmlefedkkpbbkeddpnnlc

**Comment by smplayer-dev** [comment]
It would be great (and I could probably get rid of smtube) but I don't know how to implement extensions for web browsers.

**Comment by trimechee** [comment]
There is no problem :) i have tried to change the default path in the parameters of this extension and I changed the path of VLC media player by the path of SMPlayer and it works perfectly !!
but for simplicity, it will be great if you can contact Mr Andy Portman to make a native and specific web extension for SMPlayer that rivals VLC

https://github.com/andy-portmen/open-in-vlc/

**Comment by raszpl** [comment]
Here is more direct link to the extension trimechee mentions https://github.com/andy-portmen/open-in-vlc
Personally Im not a fan of the way it works - it requires running separate nodejs client in the background performing as a man in the middle https://github.com/andy-portmen/native-client this is a 30MB full embedded chrome running all the time :(

>but I don't know how to implement extensions for web browsers

I can help with that. Iv been using my custom extension to accomplish exactly this for couple of years now.
The way I implemented mine is by registering a custom URI in Windows (similar mechanism in Linux https://unix.stackexchange.com/questions/497146/create-a-custom-url-protocol-handler)

```
[HKEY_CURRENT_USER\SOFTWARE\Classes\Magnet1]
@="Magnet URI"
"URL Protocol"=""
"Content Type"="application/x-magnet"

[HKEY_CURRENT_USER\SOFTWARE\Classes\Magnet1\shell]
@="open"

[HKEY_CURRENT_USER\SOFTWARE\Classes\Magnet1\shell\open]

[HKEY_CURRENT_USER\SOFTWARE\Classes\Magnet1\shell\open\command]
@="\"c:\\whatever\\whatever.bat\" \"%1\""
```
Yes, I just copied how Torrent client was doing it :)

Then in the browser extension you simply open "magnet1:https://www.youtube.com/watch?v=YlP4uNlrbHA" and this triggers your whatever.bat with https://www.youtube.com/watch?v=YlP4uNlrbHA as a parameter.

SMPlayer really needs https://sourceforge.net/p/smplayer/feature-requests/742/ At the very least we need a way of passing the -referrer header. A lot of video content on the web wont play without it. Ideally smplayer would just pass all the parameters it doesnt recognize to underlying player (mpv/mplayer)

**Reply by trimechee** [reply]
Very informative and relevant ! thank you very much for these ideas ans suggesntions :)

**Comment by trimechee** [comment]
Hello Mr @raszpl, my laptop breaks down so I am currently using my old 2012 windows 7 os laptop and therefore Open-in-vlc becomes important for me because the video streaming in the web browser freeze lag.....


I just asked the developer of Open-in-vlc if native client really works in the background even when you close the web browser or vlc.....


Open -in-vlc recently added e referrer and user agent string.... so the extension becomes more efficient, also I discovered the Cat-catch extension:

https://github.com/xifangczy/cat-catch



which has a "preview video" feature that we can use with vlc or potplayer but we are also supposed to use "URLProtocol" that allows developers to make custom URL protocols to launch specific applications or environment:


https://github.com/xifangczy/URLProtocol


the Cat-catch developer also advised me to add: "potplayer://${url}/referer=${referer}" in the Cat-catch preview options when I couldn't preview one video with potplayer.....
